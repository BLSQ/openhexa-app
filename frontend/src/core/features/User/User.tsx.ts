import { graphql } from "graphql/gql";

export const UserUserDoc = graphql(`
fragment User_user on User {
  id
  email
  displayName
  ...UserAvatar_user
}
`);
