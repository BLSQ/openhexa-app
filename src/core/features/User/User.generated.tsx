import * as Types from '../../../graphql-types';

import { gql } from '@apollo/client';
import { UserAvatar_UserFragmentDoc } from '../../../identity/features/UserAvatar.generated';
export type User_UserFragment = { __typename?: 'User', id: string, email: string, displayName: string, avatar: { __typename?: 'Avatar', initials: string, color: string } };

export const User_UserFragmentDoc = gql`
    fragment User_user on User {
  id
  email
  displayName
  ...UserAvatar_user
}
    ${UserAvatar_UserFragmentDoc}`;