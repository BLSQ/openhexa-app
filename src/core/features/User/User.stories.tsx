import { Story, action } from "@ladle/react";
import React from "react";
import User from "./User";

export const UserStory: Story<
  { textColor?: string; subtext?: boolean } & React.ComponentProps<
    typeof User
  >["user"]
> = ({ textColor, subtext, ...user }) => (
  <User user={user} textColor={textColor} subtext={subtext} />
);

UserStory.storyName = "default";
UserStory.args = {
  textColor: "text-teal-500",
  subtext: true,
  id: "1",
  email: "alfonsebrown@openhexa.org",
  displayName: "Alfonse Brown",
  avatar: {
    initials: "AB",
    color: "purple",
  },
};
UserStory.argTypes = {};

const defaults = { title: "core / User" };

export default defaults;
