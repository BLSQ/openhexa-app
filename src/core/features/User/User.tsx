import { gql } from "@apollo/client";
import UserAvatar from "identity/features/UserAvatar";
import { User_UserFragment } from "./User.generated";

type UserProps = {
  user: User_UserFragment;
  textColor?: string;
  subtext?: boolean;
};

const User = ({ user, subtext, textColor = "text-inherit" }: UserProps) => {
  return (
    <div className="flex items-center gap-1.5">
      <UserAvatar user={user} size="sm" />
      <div className="truncate">
        <div className={textColor}>{user.displayName}</div>
        {subtext && <div className="text-xs text-gray-400">{user.email}</div>}
      </div>
    </div>
  );
};

User.fragments = {
  user: gql`
    fragment User_user on User {
      id
      email
      displayName
      ...UserAvatar_user
    }
    ${UserAvatar.fragments.user}
  `,
};

export default User;
