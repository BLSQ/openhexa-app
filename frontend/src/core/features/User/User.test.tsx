import User from "./User";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

const USER = {
  displayName: "Test User",
  email: "testuser@openhexa.org",
  id: "UUID",
  avatar: {
    initials: "TU",
    color: "",
  },
};

describe("User", () => {
  it("renders", async () => {
    const { container } = render(<User user={USER} />);
    expect(screen.queryByText(USER.email)).toBeNull();
    expect(container).toMatchSnapshot();
  });

  it("renders with the subtext", () => {
    const { container } = render(<User user={USER} subtext />);
    expect(screen.getByText(USER.email)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
