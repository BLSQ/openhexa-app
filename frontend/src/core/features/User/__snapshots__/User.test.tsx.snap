// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User renders 1`] = `
<div>
  <div
    class="flex items-center gap-2"
  >
    <span
      class="inline-flex shrink-0 items-center justify-center rounded-full bg-gray-500 text-sm h-8 w-8"
      title="Test User"
    >
      <span
        class="font-medium leading-none text-white"
      >
        TU
      </span>
    </span>
    <div
      class="truncate"
    >
      <div
        class="text-inherit"
      >
        Test User
      </div>
    </div>
  </div>
</div>
`;

exports[`User renders with the subtext 1`] = `
<div>
  <div
    class="flex items-center gap-2"
  >
    <span
      class="inline-flex shrink-0 items-center justify-center rounded-full bg-gray-500 text-sm h-8 w-8"
      title="Test User"
    >
      <span
        class="font-medium leading-none text-white"
      >
        TU
      </span>
    </span>
    <div
      class="truncate"
    >
      <div
        class="text-inherit"
      >
        Test User
      </div>
      <div
        class="text-xs text-gray-400"
      >
        testuser@openhexa.org
      </div>
    </div>
  </div>
</div>
`;
